idaes/models/properties/modular_properties/phase_equil/smooth_VLE_2.py

```diff
@@ -0,0 +1,339 @@
+            units=f_units,
+        )
+        b.add_component("eps_z" + suffix, eps_z)
+
+        gp = Var(
+            vl_phase_set,
+            initialize=EPS_INIT,
+            bounds=(0, None),
+            doc="Slack variable for cubic second derivative for phase p",
+            units=pyunits.dimensionless,
+        )
+        b.add_component("gp" + suffix, gp)
+
+        gn = Var(
+            vl_phase_set,
+            initialize=EPS_INIT,
+            bounds=(0, None),
+            doc="Slack variable for cubic second derivative for phase p",
+            units=pyunits.dimensionless,
+        )
+        b.add_component("gn" + suffix, gn)
+
+        def rule_temperature_slack_complementarity(b, p):
+            flow_phase = b.flow_mol_phase[p]
+
+            return smooth_min(s[p] * f_units / t_units, flow_phase, eps_t) == 0
+
+        b.add_component(
+            "temperature_slack_complementarity" + suffix,
+            Constraint(
+                vl_phase_set,
+                rule=rule_temperature_slack_complementarity,
+            ),
+        )
+
+        def rule_cubic_second_derivative(b, p):
+            p1, p2 = phase_pair
+
+            _b, _, _ = calculate_equilibrium_cubic_coefficients(
+                b, cname, ctype, p1, p2, p
+            )
+            z = b.compress_fact_phase[p]
+
+            return 6 * z + 2 * _b
+
+        b.add_component(
+            "cubic_second_derivative" + suffix,
+            Expression(vl_phase_set, rule=rule_cubic_second_derivative),
+        )
+
+        def rule_cubic_root_complementarity(b, p):
+            der = getattr(b, "cubic_second_derivative" + suffix)
+            return der[p] == gp[p] - gn[p]
+
+        b.add_component(
+            "cubic_root_complementarity" + suffix,
+            Constraint(vl_phase_set, rule=rule_cubic_root_complementarity),
+        )
+
+        def rule_cubic_slack_complementarity(b, p):
+            flow_phase = b.flow_mol_phase[p]
+            if b.params.get_phase(p).is_vapor_phase():
+                return smooth_min(gn[p] * f_units, flow_phase, eps_z) == 0
+            else:
+                return smooth_min(gp[p] * f_units, flow_phase, eps_z) == 0
+
+        b.add_component(
+            "cubic_slack_complementarity" + suffix,
+            Constraint(vl_phase_set, rule=rule_cubic_slack_complementarity),
+        )
+
+    @staticmethod
+    def calculate_scaling_factors(b, phase_pair):
+        """
+        Method to calculate scaling factors for phase equilibrium
+        """
+        suffix = "_" + phase_pair[0] + "_" + phase_pair[1]
+        sf_T = iscale.get_scaling_factor(b.temperature, default=1, warning=True)
+
+        try:
+            teq_cons = getattr(b, "_teq_constraint" + suffix)
+            # pylint: disable-next=protected-access
+            iscale.set_scaling_factor(b._teq[phase_pair], sf_T)
+            iscale.constraint_scaling_transform(teq_cons, sf_T, overwrite=False)
```

Comment:
I guess we're not scaling the slacks then. If we're giving them temperature units, we might as well scale them like temperature. It would be a good first guess, at any rate, and I think the autoscalers will faceplant.

Reply:
Very well, but I do expect this case will need heuristic scaling.